added .cb_create_cohort_v2 for CBv2 endpoint

ilevantis opened the pull request:
## This PR does the following:

+ It adds the function `.cb_create_cohort_v2()` and renames the old `cb_create_cohort()` to `.cb_create_cohort_v1()`.
+ The exported function `cb_create_cohort()` now calls `.cb_create_cohort_v1()` or `.cb_create_cohort_v2()` depending on the CB version specified.

## Testing
Ensure you are configured to use the demo cohort browser instance with both CBv1 and CBv2:
```
> cloudos_whoami()
CloudOS base URL http://cohort-browser-dev-110043291.eu-west-1.elb.amazonaws.com/cohort-browser/
CloudOS token    .....
CloudOS team id  5f7c8696d6ea46288645a89f
```

```
devtools::load_all()
cb_create_cohort("ilya-r-test1", cb_version="v1")
cb_create_cohort("ilya-r-test2", cb_version="v2")
```

ilevantis commented:
R-CMD-check fails in each case with the same warning about the non-standard makefile. This is an issue we need to sort out for CRAN but not relevant to this PR.

dapineyro commented:
Just a few of comments:

For next reviews, could you please provide full instructions on how to install the package (the exact branch) and, very important, the expected output from both function calls? In that regard, it would be very useful to have a docker container already prepared to run everything and not having to deal with package installations and messing up the system. I did my own one for your PR and it's working nicely, I can share if you want.

This is the output I got after each call:

```
> cb_create_cohort("ilya-r-test1", cb_version="v1")
Cohort created successfully.
Cohort ID:  60e833cc9dd71b22256fcfda 
Cohort Name:  ilya-r-test1 
Cohort Description:   
Number of filters applied:  0 
> cb_create_cohort("ilya-r-test2", cb_version="v2")
Cohort created successfully.
Cohort ID:  60e833e09dd71b22256fcfdb 
Cohort Name:  ilya-r-test2 
Cohort Description:   
Number of filters applied:  0 
```
Is this what we expected?
In any case, I look for the new created cohorts in the cohort browser and they were created.

Next, I wanted to see what happens if I try to create an existing cohort. That was the result:

```
> cb_create_cohort("ilya-r-test1", cb_version="v1")
Error in .cb_create_cohort_v1(cohort_name = cohort_name, cohort_desc = cohort_desc,  : 
  Bad Request (HTTP 400). Failed to create a cohort.
> cb_create_cohort("ilya-r-test2", cb_version="v2")
Error in .cb_create_cohort_v2(cohort_name = cohort_name, cohort_desc = cohort_desc,  : 
  Conflict (HTTP 409). Failed to create a cohort.
```

I guess that this is expected, but not sure. In any case, an improved error message could be useful here (maybe for following PR).

Thank you Ilya and nice job!

ilevantis commented:
Thanks David. That output is as expected.

I've added extra installation information to the other pull requests and will do so for future PRs.

I have added an issue for improving the error message when trying to create a cohort with an already existing name.

sk-sahu reviewed (approved):
Looks good to me

Just an improvement suggestion #35
